### PR TITLE
fix for php 8.2  #43334

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue [#43333](https://github.com/joomla/joomla-cms/issues/43334)

### CODE REVIEW

File: src/Web/WebClient.php [hunk 1]
```diff
@@ -175,7 +175,7 @@ public function __construct($userAgent = null, $acceptEncoding = null, $acceptLa
         if (empty($userAgent) && isset($_SERVER['HTTP_USER_AGENT'])) {
             $this->userAgent = $_SERVER['HTTP_USER_AGENT'];
         } else {
-            $this->userAgent = $userAgent;
+            $this->userAgent = null === $this->userAgent ? '' : $this->userAgent;
```
Comment (HLeithner) [inline src/Web/WebClient.php]:
```suggestion
            $this->userAgent = $this->userAgent ?? '';
```

Comment (o-joomla-issues) [reply src/Web/WebClient.php]: Works here. No more PHP errors. Tested on a multilanguage website with:
```
$ openssl s_client -connect some-website.com:443
GET / HTTP/1.1[enter]
Host: some-website.com[enter]
[enter]
```

File: src/Web/WebClient.php [hunk 2]
```diff
@@ -175,7 +175,7 @@ public function __construct($userAgent = null, $acceptEncoding = null, $acceptLa
         if (empty($userAgent) && isset($_SERVER['HTTP_USER_AGENT'])) {
             $this->userAgent = $_SERVER['HTTP_USER_AGENT'];
```
Comment (Quy) [inline src/Web/WebClient.php]:
```suggestion
        if (empty($userAgent)) {
            $this->userAgent = $_SERVER['HTTP_USER_AGENT'] ?? '';
```

File: src/Web/WebClient.php [hunk 3]
```diff
@@ -175,7 +175,7 @@ public function __construct($userAgent = null, $acceptEncoding = null, $acceptLa
...
         } else {
-            $this->userAgent = $userAgent;
+            $this->userAgent = $this->userAgent ?? '';
```
Comment (Quy) [inline src/Web/WebClient.php]:
```suggestion
            $this->userAgent = $userAgent;
```

**Comment by carlovthunder** [comment]
I use Joomla 4.4.4 and Php 8.3
I have correct from:
$this->userAgent = $this->userAgent ?? '';
to            $this->userAgent = $userAgent;

The Joomla https://address/admin is malformed, it work only on http://address/admin

**Comment by rdeutz** [comment]
if my changed in https://github.com/joomla-framework/application/pull/126 will be implemented then this here is not needed. Further more setting it to am empty string doesn't make a difference to null, as far as I can see. Closing this one that for makeing the PR

**Comment by tallandtrue** [comment]
Just to add this error has cropped up for me as well on my Joomla 5.1.4 website running PHP 8.1: 

PHP Deprecated:  stripos(): Passing null to parameter #1 ($haystack) of type string is deprecated in /home/[root-folder]/[website-folder]/libraries/vendor/joomla/application/src/Web/WebClient.php

Although these are only warnings, the negative impact is that my Joomla error_log grew, which I only realised when I received disk space warnings.

For now, I've followed the advice in another post and have turned off error logging in Joomla.

**Pull request closed without merging**